lib/csetgrp.gi

```diff
@@ -788,7 +788,33 @@ local a,r;
   TryNextMethod();
 end);
 
-
+InstallMethod(Intersection2, "general groups", IsIdenticalObj,
+              [IsRightCoset,IsRightCoset],
+function(cos1,cos2)
+    local H1, H2, x1, x2, shift, sigma, theInt, cosTest, eCos, rho;
```

Comment:
I am concerned about the cases where this algorithm applies. I think there can be cases where H1 and H2 are infinite and the index is finite and that algo still works.

Does your corrections works in that case? The comparison with index also makes sense for infinite groups. But it is not sure to me how far we can go for the case of general groups as not everything is available for them.

Reply:
All bets are off when the cosets are infinite. Whether anything will work in that case will depend largely on what is implemented and how for those groups.

That said, the above check should still work fine: if both groups are infinite, it won't ever swap, and if one is finite and one is infinite, it will ensure that the finite one comes first. Sure, one could extend this to use the index of the intersection to handle things "even better", but just computing the intersection of `H1` and `H2` may already be infeasible in that case -- so it shouldn't be done before at least handling basic cases like checking whether `cos1 = cos2` etc.